handsontable/src/dataMap/metaManager/mods/__tests__/fixedColumnsProperties.unit.js

```diff
@@ -0,0 +1,169 @@
+
+describe('fixedColumnsProperties', () => {
+
+  let rtl = false;
+
+  beforeEach(() => {
+    Handsontable.mockImplementation(() => {
+      return {
+        colToProp: visualCol => `prop_${visualCol}`,
+        runHooks: () => {},
+        hasHook: () => {},
+        isRtl: () => rtl,
+      };
+    });
+  });
+
+  const dataProvider = [
+    {
+      label: 'when set nothing',
+      inRtl: false,
+      inLeft: undefined,
+      inStart: undefined,
+      throws: false,
+      expected: 0
+    },
+    {
+      label: 'when set `fixedColumnsLeft` = 0 ',
+      inRtl: false,
+      inLeft: 0,
+      inStart: undefined,
+      throws: false,
+      expected: 0
+    },
+    {
+      label: 'when set `fixedColumnsLeft` = 0 and `fixedColumnsStart` = 0 ',
+      inRtl: false,
+      inLeft: 0,
+      inStart: 0,
+      throws: true,
+      expected: undefined
+    },
+    {
+      label: 'when set `fixedColumnsStart` = 0 ',
+      inRtl: false,
+      inLeft: undefined,
+      inStart: 0,
+      throws: false,
+      expected: 0
+    },
+    {
+      label: 'when set `fixedColumnsLeft` = 1 ',
+      inRtl: false,
+      inLeft: 1,
+      inStart: undefined,
+      throws: false,
+      expected: 1
+    },
+    {
+      label: 'when set `fixedColumnsLeft` = 1 and `fixedColumnsStart` = 2 ',
+      inRtl: false,
+      inLeft: 1,
+      inStart: 2,
+      throws: true,
+      expected: undefined
+    },
+    {
+      label: 'when set `fixedColumnsStart` = 2 ',
+      inRtl: false,
+      inLeft: undefined,
+      inStart: 2,
+      throws: false,
+      expected: 2
+    },
+    {
+      label: 'when RTL, set nothing',
+      inRtl: true,
+      inLeft: undefined,
+      inStart: undefined,
+      throws: false,
+      expected: 0
+    },
+    {
+      label: 'when RTL, set `fixedColumnsLeft` = 0 ',
+      inRtl: true,
+      inLeft: 0,
+      inStart: undefined,
+      throws: true,
+      expected: undefined
+    },
+    {
+      label: 'when RTL, set `fixedColumnsLeft` = 0 and `fixedColumnsStart` = 0 ',
+      inRtl: true,
+      inLeft: 0,
+      inStart: 0,
+      throws: true,
+      expected: undefined
+    },
+    {
+      label: 'when RTL, set `fixedColumnsStart` = 0 ',
+      inRtl: true,
+      inLeft: undefined,
+      inStart: 0,
+      throws: false,
+      expected: 0
+    },
+    {
+      label: 'when RTL, set `fixedColumnsLeft` = 1 ',
+      inRtl: true,
+      inLeft: 1,
+      inStart: undefined,
+      throws: true,
+      expected: undefined
+    },
+    {
+      label: 'when RTL, set `fixedColumnsLeft` = 1 and `fixedColumnsStart` = 2 ',
+      inRtl: true,
+      inLeft: 1,
+      inStart: 2,
+      throws: true,
+      expected: undefined
+    },
+    {
+      label: 'when RTL, set `fixedColumnsStart` = 2 ',
+      inRtl: true,
+      inLeft: undefined,
+      inStart: 2,
+      throws: false,
+      expected: 2
+    },
+  ];
+
+  dataProvider.forEach(({ label, inRtl, inLeft, inStart, throws, expected }) => {
```

Comment:
Can I ask you to rewrite the tests using classic "manual" cases notation?
```js
describe('...', () => {
  it('should ...', () => {
  
  });

  it('should ...', () => {
  
  });

  it('should ...', () => {
  
  });

  // ...
});
```

IMO writing the logic within the tests is a bad idea. Tests should be written as simply as possible, they should be easy to read and modify. 

If I would like to skip executing one case (by adding `xit` or commenting the whole `it` block) it would not be easy to do that. I mean possible but with playing around with if's or so.

Reply:
If it will be acceptable for you, I propose to separate dataProvider test case which _throws / not to throws_.

Reply:
It is maybe common but in my opinion a bad practice. The object that is used here
```    {
      label: 'when RTL, set `fixedColumnsStart` = 2 ',
      inRtl: true,
      inLeft: undefined,
      inStart: 2,
      throws: false,
      expected: 2
    },
```
looks like a hack-ish method of extending the test framework. I'm trying to find some pros of this, but can't.

Reply:
I made this as you requested. 
I still conceive it is much less clear. However, it fit the team guidelines. 

Imho, it's a great topic for retro.